src/main/resources/assets/computercraft/lua/rom/programs/move.lua

```diff
@@ -8,11 +8,33 @@ end
 local sSource = shell.resolve(tArgs[1])
 local sDest = shell.resolve(tArgs[2])
 local tFiles = fs.find(sSource)
+
+local function sanityChecks(sSource)
+    if fs.isDriveRoot(sSource) then
+        printError("Can't move mounts")
+        return false
+    elseif fs.isReadOnly(sSource) then
+        printError("Source is read-only")
+        return false
+    end
+    return true
+end
+
+if fs.exists(sDest) then
+    printError("Destination exists")
+    return false
+elseif fs.isReadOnly(sDest) then
+    printError("Destination is read-only")
+    return false
+end
+
 if #tFiles > 0 then
     for _, sFile in ipairs(tFiles) do
         if fs.isDir(sDest) then
+            if not sanityChecks(sFile) then return end
```

Comment:
Hrmr, I wonder if it's better to try to copy the other files rather than aborting entirely - like we do for `rm`.